File: TrabalhosPraticos/TP_1/Questao10/palindromo.java
import java.util.Scanner;

public class palindromo{

    public static boolean endVerifier(String frase){
        
        return(frase.charAt(0) == 'F' && frase.charAt(1) == 'I' && frase.charAt(2) == 'M');
    }
    

    public static boolean IsPalindromo(int i, int n, String frase ){

        boolean test = true;
        
        if(i < n){
            if(frase.charAt(i) != frase.charAt(n)){
                test = false;
            }
            else{
                test = IsPalindromo(i+1, n-1, frase);
            }
        }

        return test;
    }


    public static void main(String[] args){

        Scanner Sc = new Scanner(System.in);

        String frase;
        boolean valor;

        frase = Sc.nextLine();

        while(!(endVerifier(frase))){
            valor = IsPalindromo(0, frase.length()-1, frase);
            if(valor == true){
                System.out.println("SIM");
            }
            else{
                System.out.println("NAO");
            }
            frase = Sc.nextLine();
        }
        Sc.close();
    }

}

File: TrabalhosPraticos/TP_1/Questao02/Palindromo.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>

bool endVerifier(char frase[]){
    bool test = true;
    if(frase[0] == 'F' && frase[1] == 'I' && frase[2] == 'M' ){
        test = false;
    }

    return test;
}

bool IsPalindrome(int i, int n, char frase[]){
    bool test = true;

    if(i < n){
        if(frase[i] != frase[n]){
            test = false;
        }
        else{
            test = IsPalindrome(i+1, n-1, frase);
        }
    }

    return test;
}


int main(){
    char frase[1000];
    bool valor;

    scanf(" %[^\n]", frase);

    while(endVerifier(frase)){
       valor = IsPalindrome(0, strlen(frase) -1, frase);
       if(valor == true){
            printf("SIM\n");
       }
       else{
        printf("NAO\n");
       }

       scanf(" %[^\n]", frase);
        
    }


    return 0;
}